src/main/java/Calculator.java

```diff
@@ -0,0 +1,61 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+public class Calculator {
+
+    private List<String> formula;
+    private double result;
+
+    // insertFormula : 계산할 공식을 계산기에 입력한다
+    public void insertFormula() {
+        Scanner scanner = new Scanner(System.in);
+        // Arrays.asList()는 고정된 사이즈를 return 하기 때문에, remove/add 와 같이 사이즈를 변경하는 작업을 진행할 수 없다.
+        // 따라서 new ArrayList<>()로 Arrays.asList()를 감싸서 고정된 길이가 아닌 가변의 길이에서 split 배열을 복사해 생성한다.
+        this.formula = new ArrayList<>(Arrays.asList(scanner.nextLine().split(" ")));
```

Comment:
```java
    // Arrays.asList() 메서드
    public static <T> List<T> asList(T... a) {
        return new ArrayList<>(a);
    }
```

Arrays.asList()의 반환 타입은 위의 List를 반환하고 있는데  "고정된 길이"라고 생각하신 이유가 궁금합니다.!
이미 List를 반환하고 있다면? ArrayList의 생성자에 넣어 객체를 생성할 필요 없이 Arrays.asList()를 그대로 formula에 초기화 할 수 있을 것 같습니다.

```suggestion
        this.formula = Arrays.asList(scanner.nextLine().split(" "))
```

Reply:
저도 처음에는 제안해주신 것과 같은 방식으로 반환된 List를 formula로 넣었습니다.  

헌데, ArrayList의 remove와 add 기능을 사용하려 하니 UnsupportedOperationException 에러가 발생했습니다.  
뭘까 싶어서 구글링을 해봤는데, asList()로 반환되는 ArrayList는 java.util.Arrays.ArrayList이고, 일반적으로 사용하는 ArrayList는 java.util.ArrayList 이더라구요.

그리고 java.util.Arrays.ArrayList는 remove를 허용하지 않는다는 것을 알았고 관련 메소드도 없다는 것을 파악할 수 있었습니다.
결정적으로는 asList() 주석에 다음과 같이 적혀져 있었습니다. 


```
Returns a fixed-size list backed by the specified array. 
(Changes to the returned list "write through" to the array.)
```

해서 new ArrayList로 리스트를 생성한 뒤 asList로 반환된 리스트를 넣어 최종적으로는 리스트의 길이를 수정하는 remove/add 메소드를 이용할 수 있게 되었습니다.

### Reference
https://stackoverflow.com/questions/43020075/java-util-arrays-aslist-when-used-with-removeif-throws-unsupportedoperationexcep

Reply:
주석에 있는 내용을 그렇게 빌드업 하셨군요.! 충분히 이해했습니다. 💯